Lib/test/test_ssl.py

```diff
@@ -4847,28 +4879,33 @@ def call_after_accept(conn_to_client):
             server_responding.set()
             return True  # Tell the server to stop.
 
+        timeout = 2.0
```

Comment:
Since this is set to `2.0`, all the changes to generalize the timeout in `SingleConnectionTestServerThread` aren't really used?

Can we at least set this to, say, `5.0` to give it some breathing room compared to the previous value?

Reply:
> Since this is set to 2.0, all the changes to generalize the timeout in SingleConnectionTestServerThread aren't really used?

I increased the timeout from 2 seconds to SHORT_TIMEOUT (at least 30 seconds) in the 2 other tests to make these tests more reliable (2 seconds may be too short on a busy system).

> Can we at least set this to, say, 5.0 to give it some breathing room compared to the previous value?

On Windows, the test takes ``timeout * 2`` seconds (4 seconds) to complete :-( I don't understand why the client doesn't fail with a timeout error as soon as the server closes its listener connection!? Right now, I prefer to fix the known issues of *dangling threads* before attempting to change **this** fragile timeout.

Reply:
it's also fair to just skip the test on windows if reliability is platform specific. the primary reproducer only happens on Linux anyways.

Reply:
The behavior on Windows is very different than on Linux. On Windows, test_https_client_non_tls_response_ignored() takes 4 seconds (server times out, then client times out). On Linux, it completes in 100 ms.

I didn't spend too much time to try to understand why/how. I prefer to continue running the test on Windows, unless there is a good reason to no do so. With my change, the test is reliable on Windows.